### PR TITLE
Refactor LICENSE.md to make it scanner friendly

### PR DESCRIPTION
**Pull request by SamuraiAku**
Most corporate environments will not approve the use of open-source software until all of the licensing terms have been reviewed and approved. Given the huge amounts of open source in use these days, companies rely on automated scanning to make a first pass at evaluation.  If the scanning software is able to comprehend the license files, then approval could be granted in just a day or two. If the scanners cannot comprehend the license file then it goes into a queue for review by a human which takes a good deal longer. Commonly this review is required for every new point release that someone at the company wants to use. Given the rapid rate of Julia development, having to go into manual review will impede adoption of the latest and greatest versions of Julia in the corporate world.
 
Currently Julia's LICENSE.md file cannot be comprehended by the scanners. A good and easy way to tell if the license is scanner friendly is by checking if Github's own automated scanners can detect the license. For further information on Github's automated scanning see:
https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/licensing-a-repository#detecting-a-license

This pull request modifies LICENSE.md to make it scanner friendly. To do that, only the top-level MIT license is listed in the file. Changes to the formatting of license text have been to make it comprehensible to the scanners but the text of the license has not been altered. The licensing for other components have been moved to a new file name named THIRDPARTY.md.

To demonstrate that the reformatted LICENSE.md can be read by license scanners, I have created a new repository that contains only this PR as it's master branch.
https://github.com/SamuraiAku/license_refactor

You can see that under "About" the page says this repository uses the MIT license. That information was automatically filled in by Github's license scanner

Please note that this PR is only one step towards making the code base full scanner friendly. Future PRs can begin to tackle the remaining issues piece by piece.

### CODE REVIEW

**Comment by ViralBShah** [comment]
The code in this repository is all MIT licensed (maybe there are some exceptions that I am forgetting), and the third party licenses probably only matter for the binaries. On the downloads page, we can link both the Julia and the third party licenses prominently.

Curious to see what others have to say.

**Comment by SamuraiAku** [comment]
Following the text in license.md, the following source files are covered by a different license than MIT

- src/crc32.c : zlib license; credit (although not the license) is given in the source file
- src/ccall.cpp starting at line 288 : bsd-3 license
- some portion of src/jitlayers.cpp : uiuc
- src/disarm.cpp: uiuc license
- src/getopt.c: MIT license from another project (so I guess it doesn’t have to be listed in the license file?)
- some code from MINGW project for dirname on windows ; multiple licenses listed on homepage; Julia lists MIT
- Some other functions from netbsd; bsd-3 license
- Some functions from python; Python license
- Some code from google for cyclecount; Apache license
- everything in base/ryu; Boost license
- Everything in base/grisu; except that folder doesn’t exist? Should that line be removed?
- base/src/{exp, rem_pio2, hyperbolic}.jl; exp.jl is missing a copyright notice in the file. Portions of the other two files are covered by FDLIBM license

**Comment by oscardssmith** [comment]
`exp` and all the variants are mit from 1.6 on. I rewrote the entire file from scratch using a different algorithm.

**Comment by SamuraiAku** [comment]
> `exp` and all the variants are mit from 1.6 on. I rewrote the entire file from scratch using a different algorithm.

Good to know. That’s something to be removed from the license files in another PR.

**Comment by vtjnash** [comment]
Yeah, seems to need cleanup. I don't think cyclecount should have ever been listed and grisu is deleted.

**Comment by SamuraiAku** [comment]
I can open another PR for thirdparty cleanup after this one is dispositioned.

**Comment by ViralBShah** [comment]
@StefanKarpinski Any thoughts here? We should probably keep this open a bit longer to give time to folks to comment.

**Comment by oscardssmith** [comment]
Perhaps we should just add it to triage?

**Comment by StefanKarpinski** [comment]
I think it's fairly important to leave some kind of breadcrumb for someone who is looking at the license file indicating that there may be other licensing considerations when using the result of compiling this code. If an extra line at the bottom could link to the `THIRDPARTY.md` file that would be fine.

**Comment by SamuraiAku** [comment]
I think that will do it. At first adding the pointer to THIRDPARTY.md broke the license scanner until I found a way to make it ignore  everything after the license.

https://github.com/licensee/licensee/issues/498

File: THIRDPARTY.md
Comment (StefanKarpinski) [inline THIRDPARTY.md:1]:
```suggestion
The Julia language is licensed under the MIT License (see `LICENSE.md`). The "language" consists
```
Good to leave a xref back to the license file here (and rewrap the lines).

**Review by StefanKarpinski — APPROVED** [review]
This is looking good to me. With cross references between the two files this seems clear and it's good that it helps automatic license detection do its thing.

**Comment by ViralBShah** [comment]
Did triage discuss this?

**Comment by oscardssmith** [comment]
We didn't have triage last week. (So no)

File: LICENSE.md
Comment (JeffBezanson) [inline LICENSE.md:26]:
```suggestion
Please see THIRDPARTY.md for license information for other software used in this project.
```

**Comment by JeffBezanson** [comment]
Triage says yes.

**Comment by mbauman** [comment]
There doesn't seem to be a definitive idiom for this, but this [SoftwareEngineering.SE question its answers](https://softwareengineering.stackexchange.com/questions/304874/declaring-multiple-licences-in-a-github-project) suggest some similar ideas — it seems as long as whatever we do here is obvious we're in good stead.  IMO most options suggested there are _less_ obvious than this.

**Comment by SamuraiAku** [comment]
@ViralBShah Does this PR need your approval before it can move forward?

**Review by ViralBShah — APPROVED** [review]

**Comment by ViralBShah** [comment]
What's going on in the CI that is preventing the merging? I think this is good to go.

File: LICENSE.md
Comment (ViralBShah) [inline LICENSE.md:26]: It would be nice to link the URL of THIRDPARTY.md in here (either now or after merging).

**Comment by SamuraiAku** [comment]
It looks like maybe the version that I made these changes to had other issues. Should I rebase?

**Comment by ViralBShah** [comment]
Sure no harm rebasing

**Comment by SamuraiAku** [comment]
@ViralBShah I’ve  rebased and now CI is passing. This should be enough to move forward right?

**Comment by SamuraiAku** [comment]
@ViralBShah can you add the back port labels then? Thanks.

**Pull request merged**
